src/jit/flowgraph.cpp

```diff
@@ -10228,17 +10216,12 @@ void                Compiler::fgUnreachableBlock(BasicBlock* block)
 
     if (block->IsLIR())
     {
-        LIR::Range blockRange = LIR::AsRange(block);
-
-        // Skip phis when decrementing ref counts.
-        GenTree* firstNonPhiNode = blockRange.FirstNonPhiNode();
-        if (firstNonPhiNode != blockRange.End())
+        LIR::Range& blockRange = LIR::AsRange(block);
+        if (!blockRange.IsEmpty())
         {
-            LIR::DecRefCnts(this, block, LIR::AsRange(firstNonPhiNode, blockRange.EndExclusive()));
+            LIR::DecRefCnts(this, block, blockRange.NonPhiNodes());
+            blockRange.Remove(blockRange.FirstNode(), blockRange.LastNode());
         }
```

Comment:
No, because of move semantics. The call would look like `blockRange.Remove(std::move(blockRange))`. The call to `std::move` would invalidate `blockRange` and `Remove` would be called on an uninitialized value.